app/forms/register/SchemeDetailsFormProvider.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.register
+
+import javax.inject.Inject
+import forms.mappings.{Constraints, Mappings}
+import play.api.data.Form
+import play.api.data.Forms._
+import models.{SchemeDetails, SchemeType}
+
+class SchemeDetailsFormProvider @Inject() extends Mappings with Constraints {
+
+  def apply(): Form[SchemeDetails] = Form(mapping(
+    "schemeName" -> text(
+      "schemeDetails.schemeName.error.required").
+      verifying(maxLength(255, "schemeDetails.schemeName.error.length")),
+    "schemeType" -> schemeTypeMapping(
```

Comment:
Could these keys be set as defaults in the `schemeTypeMapping` method?

I think it's good that they're overridable but it's likely that we'd never actually want to change them from the default.